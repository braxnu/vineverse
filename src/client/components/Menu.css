body {
    margin: 0;
}
.mobile-nav {
    width: 100%;
    height: 60px;
    position: absolute;
    bottom: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: space-around;
}
