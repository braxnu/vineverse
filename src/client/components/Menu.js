import React from 'react'
import './Menu.css'

const Menu = props => {
  const itemList = props.items.map(item => (
    <a href={item.url} key={item.display}>
      <div>{item.display}</div>
    </a>
  ))
  return (props.screenSize === 'mobile') ?
    (<div className="mobile-nav">{itemList}</div>) :
    (<div className="desktop-nav">{itemList}</div>)
}
export default Menu