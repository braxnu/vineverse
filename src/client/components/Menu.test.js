import React from 'react'
import Menu from './Menu'
import Enzyme, { mount } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

Enzyme.configure({ adapter: new Adapter() })

it('renders a button', () => {
  const items = [
    {url: '/adin', display: 'Adin'},
    {url: '/dwa', display: 'Dwa'},
    {url: '/adin', display: 'Adin2'},
    {url: '/dwa', display: 'Dwa2'},
    {url: '/adin', display: 'Adin3'},
    {url: '/dwa', display: 'Dwa3'},
    {url: '/adin', display: 'Adin4'},
    {url: '/dwa', display: 'Dwa4'},
  ]

  const wrapper = mount(
    <Menu
      items={items}
      screenSize='mobile'
    />
  )

  expect(wrapper.html()).toMatchSnapshot()
})

it('renders all provided items', () => {
  const items = [
    {url: '/adin', display: 'Adin'},
    {url: '/dwa', display: 'Dwa'},
    {url: '/adin', display: 'Adin2'},
    {url: '/dwa', display: 'Dwa2'},
    {url: '/adin', display: 'Adin3'},
    {url: '/dwa', display: 'Dwa3'},
    {url: '/adin', display: 'Adin4'},
    {url: '/dwa', display: 'Dwa4'},
  ]

  const wrapper = mount(
    <Menu
      items={items}
      screenSize='mobile'
    />
  )

  expect(wrapper.find('a').length).toBe(items.length)
})
