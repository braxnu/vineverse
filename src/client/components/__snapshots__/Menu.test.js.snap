// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button 1`] = `"<div class=\\"mobile-nav\\"><a href=\\"/adin\\"><div>Adin</div></a><a href=\\"/dwa\\"><div>Dwa</div></a><a href=\\"/adin\\"><div>Adin2</div></a><a href=\\"/dwa\\"><div>Dwa2</div></a><a href=\\"/adin\\"><div>Adin3</div></a><a href=\\"/dwa\\"><div>Dwa3</div></a><a href=\\"/adin\\"><div>Adin4</div></a><a href=\\"/dwa\\"><div>Dwa4</div></a></div>"`;
